x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/log_retention_logic.test.ts

```diff
@@ -0,0 +1,370 @@
+    logsRetentionUpdating: false,
+  };
+
+  const mount = (defaults?: object) => {
+    if (!defaults) {
+      resetContext({});
+    } else {
+      resetContext({
+        defaults: {
+          enterprise_search: {
+            app_search: {
+              log_retention_logic: {
+                ...defaults,
+              },
+            },
+          },
+        },
+      });
+    }
+    LogRetentionLogic.mount();
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(LogRetentionLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('setOpenModal', () => {
+      describe('openModal', () => {
+        it('should be set to the provided value', () => {
+          mount();
+
+          LogRetentionLogic.actions.setOpenModal(ELogRetentionOptions.Analytics);
+
+          expect(LogRetentionLogic.values).toEqual({
+            ...DEFAULT_VALUES,
+            openModal: ELogRetentionOptions.Analytics,
+          });
+        });
+      });
+    });
+
+    describe('closeModals', () => {
+      describe('openModal', () => {
+        it('resets openModal to null', () => {
+          mount({
+            openModal: 'analytics',
+          });
+
+          LogRetentionLogic.actions.closeModals();
+
+          expect(LogRetentionLogic.values).toEqual({
+            ...DEFAULT_VALUES,
+            openModal: null,
+          });
+        });
+      });
+
+      describe('logsRetentionUpdating', () => {
+        it('resets logsRetentionUpdating to false', () => {
+          mount({
+            logsRetentionUpdating: true,
+          });
+
+          LogRetentionLogic.actions.closeModals();
+
+          expect(LogRetentionLogic.values).toEqual({
+            ...DEFAULT_VALUES,
+            logsRetentionUpdating: false,
+          });
+        });
+      });
+    });
+
+    describe('clearLogRetentionUpdating', () => {
+      describe('logsRetentionUpdating', () => {
+        it('resets logsRetentionUpdating to false', () => {
+          mount({
+            logsRetentionUpdating: true,
+          });
+
+          LogRetentionLogic.actions.clearLogRetentionUpdating();
+
+          expect(LogRetentionLogic.values).toEqual({
+            ...DEFAULT_VALUES,
+            logsRetentionUpdating: false,
+          });
+        });
+      });
+    });
+
+    describe('updateLogRetention', () => {
+      describe('logRetention', () => {
+        it('updates the logRetention values that are passed, and defaults the others that are not set', () => {
+          mount({
+            logRetention: {},
+          });
+
+          LogRetentionLogic.actions.updateLogRetention({
+            api: { enabled: true },
+          });
+
+          expect(LogRetentionLogic.values).toEqual({
+            ...DEFAULT_VALUES,
+            logRetention: {
+              api: { enabled: true },
+              analytics: {}, // This is defaulted to {}
```

Comment:
I think I missed this sorry, where are we defaulting this?

Reply:
I don't know. I simply wrote a test for existing behavior.

Reply:
The original code base had the api and analytics options as optional on the ILogRetention type, specifically for this method. However, this is only invoked using responses from the server as parameters, and they are always fully formed. I simplified this to just be a straight forward set value operation: https://github.com/elastic/kibana/pull/82364/commits/9788589ad094396bce0ce83870c0165b4c7d6293

Reply:
👍 I'm good with that logic change! If we simply rely on the data coming in from the server, that does seem easier & simpler. It is a pretty significant change though so just going to CC @byronhulcher in here for info (after he comes back from OOO) / to see if he knows something we don't